functests/utils/pods/pods.go

```diff
@@ -121,6 +123,14 @@ func WaitForPhase(pod *corev1.Pod, phase corev1.PodPhase, timeout time.Duration)
 
 		return false, nil
 	})
+	if err != nil {
+		updatedPod := &corev1.Pod{}
+		if err := testclient.Client.Get(context.TODO(), key, updatedPod); err != nil {
+			testlog.Errorf("could not get the pod's object: %v", err)
+		}
+		testlog.Errorf("Expected the pod to reach phase %s but its phase is %s", phase, updatedPod.Status.Phase)
+	}
+	return err
```

Comment:
I did this change in WaitForPhase and not WaitForPredicate because in the predicate:
1. we need to pass a pred function, which will be actually the same functionality we are checking in WaitForPhase, so avoid duplicated code for the pred function.
2. the Function can be checking anything so pointing to the failing condition of pod's phase in WaitForPhase is more clear.